rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,158 @@
+    `master` in parallel with a release stabilization.
+11. During stabilization, the release manager should update
+    `nixUnstable` to point to various commits of the `#-stabilization`
+    branch, to easily enable wider testing and integration.
+12. When the release manager determines readiness, the relevant commit
+    is announced to the community. Barring objections or last-minute
+    fixes judged valid by the release manager (or core team), the
+    commit is tagged and the branch receives no further commits.
+    Normal development against `master` continues. Readiness is a
+    judgment call, and should require increased scrutiny/validation
+    for a release with more complex/major changes.
+13. Associated with each `NixOS` stable release is an associated
+    `maintained` series of `Nix`, with the same support lifetime as
+    the stable release.
+14. If `master` has not yet diverged enough to warrant a minor semver
+    bump relative to the latest release of a `maintained` series, a
+    maintenance release can follow the normal stabilization workflow.
+15. If `master` has diverged too much, a `#-maintenance` branch (e.g.
+    `2.1-maintenance` can be created.
+16. Any important bugfix can be cherry-picked from `master` to a
+    `#-maintenance` branch at any committer's discretion.
+17. In rare cases, if a change truly only makes sense on a
+    `#-maintenance` branch (e.g. when we added support to `1.11` to
+    be able to work with a `2.0` database schema) it can be targeted
+    directly to the `#-maintenance` branch. This should be avoided if
+    possible and subject to especially careful testing and review.
+18. A maintenance release can be tagged off of the `#-maintenance`
+    branch at any time *after* there is a mainline release that
+    contains all of the fixes included in the desired maintenance
+    release, at the discretion of any Nix core team member.
+19. \*Automated testing, ideally applied to all mainline PRs, should
+     ensure that all releases of all supported maintained series are
+     able to successfully install master if a release were cut. This
+     should include a number of definitions of "install", especially
+     `NixOS` upgrades.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback to having a formalized model in general is our model
+may not match a reasonable productive process in practice and so may
+lead to obnoxious busy work or delays for no purpose.
+
+The drawbacks of this specific model are a bit hard to see without
+putting it into practice. One thing that seems apparent is there is a
+non-trivial amount of complexity (19 bullet points!) which could
+perhaps be reduced. It also may prove difficult under this model to
+develop especially large features, as without the right development
+practices (e.g. feature flags on `master`) this can tend to create
+new long-lived branches that can effectively become the new `master`.
```

Comment:
Maybe add to the future work: guidelines for developing major new features without long-lived unmerged branches, but also without damaging the master branch stability?

Reply:
Added, thanks!